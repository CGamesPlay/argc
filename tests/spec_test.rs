#[test]
fn test_spec_help() {
    snapshot!(include_str!("spec.sh"), &["spec", "-h"]);
}

#[test]
fn test_spec_cmd_prefered_help() {
    snapshot!(include_str!("spec.sh"), &["spec", "cmd-prefered", "-h"]);
}

#[test]
fn test_spec_cmd_omitted_help() {
    snapshot!(include_str!("spec.sh"), &["spec", "cmd-omitted", "-h"]);
}

#[test]
fn test_spec_cmd_option_names_help() {
    snapshot!(include_str!("spec.sh"), &["spec", "cmd-option-names", "-h"]);
}

#[test]
fn test_spec_cmd_option_formats_help() {
    snapshot!(
        include_str!("spec.sh"),
        &["spec", "cmd-option-formats", "-h"]
    );
}

#[test]
fn test_spec_cmd_option_quotes_help() {
    snapshot!(
        include_str!("spec.sh"),
        &["spec", "cmd-option-quotes", "-h"]
    );
}

#[test]
fn test_spec_cmd_flag_formats_help() {
    snapshot!(include_str!("spec.sh"), &["spec", "cmd-flag-formats", "-h"]);
}

#[test]
fn test_spec_cmd_positional_only_help() {
    snapshot!(
        include_str!("spec.sh"),
        &["spec", "cmd-positional-only", "-h"]
    );
}

#[test]
fn test_spec_cmd_positional_requires_help() {
    snapshot!(
        include_str!("spec.sh"),
        &["spec", "cmd-positional-requires", "-h"]
    );
}

#[test]
fn test_spec_cmd_prefered_exec() {
    snapshot!(
        include_str!("spec.sh"),
        &["spec", "cmd-prefered", "-f", "-o", "A", "AB", "C D"]
    );
}

#[test]
fn test_spec_cmd_option_names_exec() {
    snapshot!(
        include_str!("spec.sh"),
        &[
            "spec",
            "cmd-option-names",
            "--opt2",
            "value2",
            "--opt3",
            "value3_0,value3_1",
            "--opt4",
            "value4_0",
            "--opt4",
            "value4_1",
            "--opt6",
            "a",
        ]
    );
}
