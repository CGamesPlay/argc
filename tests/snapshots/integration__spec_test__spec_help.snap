---
source: tests/spec_test.rs
assertion_line: 3
expression: output

---
RUN
spec -h

STDOUT


STDERR
spec 
Test all aspects

USAGE:
    spec <SUBCOMMAND>

OPTIONS:
    -h, --help    Print help information

SUBCOMMANDS:
    cmd-flag-formats           All kindof flags
    cmd-omitted                Omitted
    cmd-option-formats         Optiona all kindof formats
    cmd-option-names           Options all kindof names
    cmd-option-quotes          Option value quoted
    cmd-positional-only        Poistional One required
    cmd-positional-requires    Poistional All required
    cmd-prefered               Preferred
    help                       Print this message or the help of the given subcommand(s)


